### PR TITLE
python310Packages.array-record: 0.4.1 -> 0.5.0

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

Changelog: https://github.com/google/array_record/releases/tag/v0.5.0
Fix `array-record` and also make it available for python 3.9 and 3.11.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/array-record/default.nix [hunk 1]
```diff
@@ -1,27 +1,36 @@
...
 }:
 
 buildPythonPackage rec {
   pname = "array-record";
-  version = "0.4.1";
+  version = "0.5.0";
   format = "wheel";
 
-  disabled = python.pythonVersion != "3.10";
+  disabled = pythonOlder "3.9";
```
Comment (samuela) [inline pkgs/development/python-modules/array-record/default.nix:18]:
```suggestion
  # As of 2023-10-31, PyPI includes wheels for Python 3.9, 3.10, and 3.11.
  disabled = pythonOlder "3.9";
```
not sure what the best way to implement that version constraint is off the top of my head, but you get the idea

Comment (GaetanLepage) [reply pkgs/development/python-modules/array-record/default.nix:19]: I just added the comment.

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 264634` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.dm-sonnet</li>
    <li>python310Packages.dm-sonnet.dist</li>
    <li>python310Packages.tensorflow-datasets</li>
    <li>python310Packages.tensorflow-datasets.dist</li>
    <li>python311Packages.dm-sonnet</li>
    <li>python311Packages.dm-sonnet.dist</li>
    <li>python311Packages.tensorflow-datasets</li>
    <li>python311Packages.tensorflow-datasets.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.array-record</li>
    <li>python310Packages.array-record.dist</li>
    <li>python311Packages.array-record</li>
    <li>python311Packages.array-record.dist</li>
  </ul>
</details>

**Review by natsukium — CHANGES_REQUESTED** [review]
Please change the commit message and the PR title to `python310Packages.array-record: 0.4.1 -> 0.5.0`.

**Review by natsukium — APPROVED** [review]
LGTM

**Pull request merged**